terraform/terraform-init.sh

```diff
@@ -97,10 +98,26 @@ azure_location_to_shortloc () {
 
 init_state_storage () {
     echo "[+] Initializing state storage"
+    # Assign variables STATE_RG and STATE_ACCOUNT: these variables are
+    # used to select the remote state storage used in this instance.
+    if [ "$SHORTLOC" = "eun" ]; then
+        # State file for 'eun' is named differently to not have to move
+        # all existing state files to new 'ghaf-infra-state-eun' resource
+        # group from the one currently used (see below).
+        # At some point we might want to rename the 'eun' state also,
+        # but it would require manual actions to import existing environments.
+        STATE_RG="ghaf-infra-state"
+        STATE_ACCOUNT="ghafinfratfstatestorage"
+    else
+        STATE_RG="ghaf-infra-state-$SHORTLOC"
+        STATE_ACCOUNT="ghafinfrastate$SHORTLOC"
+    fi
+    echo -e "storage_account_rg_name=$STATE_RG\nstorage_account_name=$STATE_ACCOUNT" >"$MYDIR/.env"
     # See: ./state-storage
     pushd "$MYDIR/state-storage" >"$OUT"
     terraform init >"$OUT"
-    if ! terraform apply -auto-approve &>"$OUT"; then
+    terraform workspace select "$SHORTLOC" &>"$OUT" || terraform workspace new "$SHORTLOC" >"$OUT"
+    if ! terraform apply -var="location=$LOCATION" -auto-approve &>"$OUT"; then
```

Comment:
for instance here

Reply:
In this example, we want to continue in case the apply fails, and assume the failure is due to already existing state storage which we simply output inside the if, then continue with the script execution. If that assumption later turns out to be incorrect, the script will fail some point later.

You can see the error messages from such 'accepted' failure cases if you run the script with `-v` option.